pkg/alertmanager/alertstore/bucketclient/bucket_client.go

```diff
@@ -0,0 +1,110 @@
+package bucketclient
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/objstore"
+	"github.com/thanos-io/thanos/pkg/runutil"
+
+	"github.com/cortexproject/cortex/pkg/alertmanager/alertspb"
+	"github.com/cortexproject/cortex/pkg/storage/bucket"
+)
+
+const (
+	// The bucket prefix under which all tenants alertmanager configs are stored.
+	alertsPrefix = "alerts"
+)
+
+// BucketAlertStore is used to support the AlertStore interface against an object storage backend. It is implemented
+// using the Thanos objstore.Bucket interface
+type BucketAlertStore struct {
+	bucket      objstore.Bucket
+	cfgProvider bucket.TenantConfigProvider
+	logger      log.Logger
+}
+
+func NewBucketAlertStore(bkt objstore.Bucket, cfgProvider bucket.TenantConfigProvider, logger log.Logger) *BucketAlertStore {
+	return &BucketAlertStore{
+		bucket:      bucket.NewPrefixedBucketClient(bkt, alertsPrefix),
+		cfgProvider: cfgProvider,
+		logger:      logger,
+	}
+}
+
+// ListAlertConfigs implements alertstore.AlertStore.
+func (s *BucketAlertStore) ListAlertConfigs(ctx context.Context) (map[string]alertspb.AlertConfigDesc, error) {
+	cfgs := map[string]alertspb.AlertConfigDesc{}
+
+	err := s.bucket.Iter(ctx, "", func(key string) error {
+		userID := key
+
+		cfg, err := s.getAlertConfig(ctx, userID)
+		if err != nil {
+			return errors.Wrapf(nil, "failed to fetch alertmanager config for user %s", userID)
```

Comment:
I think `Wrapf(nil, ...`) will always return `nil` here - https://pkg.go.dev/github.com/pkg/errors#Wrapf

Did you mean `Errorf()` here?

Reply:
Damn, good job spotting it! I meant `Wrapf(err, ...)`.